rpc/filters.go

```diff
@@ -115,11 +116,38 @@ func (f *Filter) pollForBlocks() error {
 	}
 }
 
+func (f *Filter) pollForTransactions() error {
+	for {
+		if f.stopped {
+			return nil
+		}
+
+		txs, err := f.backend.PendingTransactions()
+		if err != nil {
+			return err
+		}
+
+		for _, tx := range txs {
+			f.hashes = append(f.hashes, tx.Hash)
+		}
```

Comment:
this allows for duplicate transactions to be put into the filter result, maybe change to map?

Reply:
Sure, but that is also the case for the block and other funcs. I suggest we do this as part of a refactor on the filters that should include other funcs. 
For now I added a contains method that should solve it.

Reply:
it's not the case for the block filter, since the block filter only adds a block when there is a new block created